src/script/common/c_content.cpp

```diff
@@ -1959,8 +1959,28 @@ void push_objectRef(lua_State *L, const u16 id)
 
 void read_hud_element(lua_State *L, HudElement *elem)
 {
-	elem->type = (HudElementType)getenumfield(L, 2, "hud_elem_type",
-									es_HudElementType, HUD_ELEM_TEXT);
+	std::string type_string;
+	bool has_type = getstringfield(L, 2, "type", type_string);
+
+	// Handle deprecated hud_elem_type
+	std::string deprecated_type_string;
+	if (getstringfield(L, 2, "hud_elem_type", deprecated_type_string)) {
+		if (has_type && deprecated_type_string != type_string) {
+			script_log_unique(L, "Ambiguous HUD element fields \"type\" and \"hud_elem_type\", "
+					"\"type\" will be used.", warningstream);
+		} else {
+			has_type = true;
+			type_string = deprecated_type_string;
+			script_log_unique(L, "Deprecated \"hud_elem_type\" field, use \"type\" instead.",
```

Comment:
we have `log_deprecated` for this

Reply:
It was `log_deprecated` before, but I changed it to `script_log_unique` because SmallJoker wrote:
> This function might be called multiple times per server step. Please eliminate duplicates using `script_log_unique`.

Do we need a new function for this, e.g. `log_deprecated_unique`?

Reply:
`log_deprecated` can do that since very recently: cad8e895f24c74d4cf329594e7069797f8480602

Reply:
OK, it now uses log_deprecated.
(I had to rebase, and tested it, It works fine.)